Unable to run make-deploy dev with go version 1.13

/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
Run `make deploy-dev` and it fails with error
```
2021/04/30 13:05:44 Using base gcr.io/distroless/static:latest for github.com/kubeflow/kfserving/cmd/agent
2021/04/30 13:05:45 Building github.com/kubeflow/kfserving/cmd/agent
2021/04/30 13:06:28 Unexpected error running "go build": exit status 2
...
../../../pkg/mod/knative.dev/serving@v0.18.0/pkg/http/header.go:25:13: header.Values undefined (type http.Header has no field or method Values)
note: module requires Go 1.14
```
On further inspection I notice that on version 0.18.0 they[ introduced a change](https://github.com/knative/serving/commit/840ef864296681809208f9f9680db65dd78ae817) that uses a new [https/http header feature introduced in go 1.14](https://golang.org/pkg/net/http/#Header.Values).

**What did you expect to happen:**
Based on the [developer's guide](https://github.com/kubeflow/kfserving/blob/master/docs/DEVELOPER_GUIDE.md#install-requirements) golang version 1.13 should be fine to use. `Make deploy-dev` should work as expected.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- KFServing Version: Latest
- Kubernetes version: (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-30T20:19:45Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

--- a/docs/DEVELOPER_GUIDE.md
+++ b/docs/DEVELOPER_GUIDE.md
@@ -39,7 +39,7 @@ Before submitting a PR, see also [CONTRIBUTING.md](../CONTRIBUTING.md).
 
 You must install these tools:
 
-1. [`go`](https://golang.org/doc/install): KFServing controller is written in Go and requires Go 1.13.0+.
+1. [`go`](https://golang.org/doc/install): KFServing controller is written in Go and requires Go 1.14.0+.
 1. [`git`](https://help.github.com/articles/set-up-git/): For source control.
 1. [`Go Module`](https://blog.golang.org/using-go-modules): Go's new dependency management system.
 1. [`ko`](https://github.com/google/ko):

